Mathlib/Topology/UniformSpace/CompactConvergence.lean

```diff
@@ -223,6 +221,29 @@ theorem mem_compactConvergence_entourage_iff (X : Set (C(α, β) × C(α, β)))
   simp [hasBasis_compactConvergenceUniformity.mem_iff, and_assoc]
 #align continuous_map.mem_compact_convergence_entourage_iff ContinuousMap.mem_compactConvergence_entourage_iff
 
+theorem _root_.CompactExhaustion.hasBasis_compactConvergenceUniformity {ι : Type*}
+    {p : ι → Prop} {V : ι → Set (β × β)} (K : CompactExhaustion α) (hb : (𝓤 β).HasBasis p V) :
+    HasBasis (𝓤 C(α, β)) (fun i : ℕ × ι ↦ p i.2) fun i ↦
+      {fg | ∀ x ∈ K i.1, (fg.1 x, fg.2 x) ∈ V i.2} := by
+  refine hb.compactConvergenceUniformity.to_hasBasis ?_ fun i hi ↦
+    ⟨(K i.1, i.2), ⟨K.isCompact _, hi⟩, Subset.rfl⟩
+  rintro ⟨L, i⟩ ⟨hL, hi⟩
+  rcases K.exists_isCompact_subset hL with ⟨n, hn⟩
+  exact ⟨(n, i), hi, fun _fg h x hx ↦ h x <| hn hx⟩
+
+/-- If `α` is a weakly locally compact σ-compact space
+(e.g., a proper pseudometric space or a compact spaces)
+and the uniformity on `β` is pseudometrizable,
+then the uniformity on `C(α, β)` is pseudometrizable too.
+
+TODO: prove a corollary in terms of pseudometrizable topologies,
+maybe after we redefine it without `MetricSpace`s, see #2032. -/
```

Comment:
Do you mean that it will become easier with this? I don't see why, in fact I think it would be fairly straightforward in the current situation? Am I missing something obvious?

Reply:
We won't need to `import` metric spaces for that result. I can add a temporary file that `import`s enough and prove metrizability there.

Reply:
Ah, I didn't realize that was an import issue. Yes I think it would be best to add the `PseudoMetrizableSpace` result already in another file, if that's fine for you.

Reply:
Done.